### ISSUE TITLE
XML output: "extract" field has extra leading and trailing characters compared to old W3C checker

### ISSUE THREAD

**Issue by rujiel** [validator/validator#837]
The "extract" field, which contains the sample of HTML that is at fault, has some extra leading contect. I am wondering if this is intended or configurable. If I validate this doc, for example:

```
<!DOCTYPE html>
<html>
<head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="Description" content="extra characters">
</head>
<body>hi</body>
</html>
```

The validator description and extract comes back as such:

```
(Element head is missing a required instance of child element title.)
racters">
</head>
<body
```

Note how the last 10 characters of the meta tag are included. (It's always an extra 10 leading characters.) But if I send the same doc to the old w3c checker, I get back a much smaller snippet:

```
(Element head is missing a required instance of child element title.)
</head>
```

One other thing I noticed that I wonder if it is related: the old checker would highlight the exact character location of the problem, like this for the example above (closing caret for the <head> tag):

`<strong title="Position where error was detected.">&gt;</strong>`

I only noticed this because the previous version of the achecker (before the Nu validator) does not have the added leading characters, and highlights the problem character in `<strong>` tags.

Sorry for my flurry of filings, I think this will be the last one. Edit: nevermind about the trailing characters, only leading ones.

**Comment by sideshowbarker** [comment]
The leading characters are intended behavior. Providing more context is better, not worse.

> One other thing I noticed that I wonder if it is related: the old checker would highlight the exact character location of the problem

The HTML checker XML output format provides the `m` element:

https://github.com/validator/validator/wiki/Output-%C2%BB-XML#the-extract-element

> The `m` element represents a highlighted part of the extract
that pinpoints the source position associated with the message. The `m`
element contains the highlighted part of the text.
